### PR TITLE
[WIP] PHPC-316: config.w32 should define MONGOC_ENABLE_SASL

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPC-316

### CODE REVIEW

File: config.m4 [hunk 1]
```diff
@@ -406,6 +406,12 @@ if test "$PHP_MONGODB_SASL" != "no"; then
   ], [
     -L$MONGODB_SASL_DIR/$PHP_LIBDIR
   ])
+
+  PHP_CHECK_LIBRARY(sasl2, sasl_client_done,
+  [
+    MONGOC_HAVE_SASL_CLIENT_DONE=1
+    AC_SUBST(MONGOC_HAVE_SASL_CLIENT_DONE)
+  ])
```
Comment (jmikola) [inline config.m4]: I noticed elsewhere in `config.m4` we split the constant definition and `AC_SUBST` into separate statements. Is there any reason we don't simplify as `AC_SUBST(MONGOC_HAVE_SASL_CLIENT_DONE, 1)`?

Comment (bjori) [inline config.m4]: No idea. I imagine version dependencies? No prior art in php-src makes the use of it "risky"

File: config.w32 [hunk 2]
```diff
@@ -20,9 +20,14 @@ if (PHP_MONGODB != "no") {
 
   AC_DEFINE("HAVE_MONGOC", 1);
 
-  if (!CHECK_LIB("libsasl.lib", "mongodb", PHP_MONGODB)) {
-	  WARNING("mongodb libsasl support not enabled, libs not found");
+  if (PHP_MONGODB_SASL != "no") {
```
Comment (jmikola) [inline config.w32]: This is based on what we were doing for 1.x [here](https://github.com/mongodb/mongo-php-driver/blob/239a4324911094ff6c755afaf78aadc3981194b1/config.w32).

File: config.w32 [hunk 3]
```diff
@@ -20,9 +20,14 @@ if (PHP_MONGODB != "no") {
...
+    if (CHECK_LIB("libsasl.lib", "mongodb", PHP_MONGODB)) {
+      ADD_FLAG("CFLAGS_MONGODB", " /D MONGOC_ENABLE_SASL=1");
```
Comment (jmikola) [inline config.w32]: In this case, I think we still need to check for `sasl_client_done()` (as we do in `config.m4` when calling `PHP_CHECK_LIBRARY(sasl2, sasl_client_done, ...)`). I'm currently looking into whether `CHECK_FUNC_IN_HEADER()` can be used.

**Comment by bjori** [comment]
If it works on windows, lgtm

**Comment by jmikola** [comment]
I'll revisit this once I can test the make build/test targets with your Windows VM.

**Comment by bjori** [comment]
Since the Windows build instructions have very specific dependencies I don't believe config.w32 generally contains as complex checks as config.m4, especially when it comes to version/function existence.

I'm not sure about the PHPC-316 commit, but the PHPC-306 is good and should maybe be merged standalone?

**Comment by jmikola** [comment]
I split off the `config.m4` change into #55 and left this PR for PHPC-316 (despite its branch name) since we had relevant comments above. We can revisit this when the Windows build environment is ready.

**Comment by jmikola** [comment]
This PR should be obsolete. A proper check for `MONGOC_ENABLE_SASL` was implemented in https://github.com/mongodb/mongo-php-driver/pull/276 for [PHPC-629](https://jira.mongodb.org/browse/PHPC-629).

**Pull request closed without merging**